lib/log/internal/sinkcores/sentrycore/worker.go

```diff
@@ -0,0 +1,155 @@
+package sentrycore
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	// flushDelay defines the brief window in which log messages are still accepted
+	// before flushing.
+	flushDelay = 500 * time.Millisecond
+	// sentryTimeout defines how much time Sentry has to send the events.
+	sentryTimeout = 5 * time.Second
+)
+
+// worker encapsulate the process of sending events to Sentry by asynchronously
+// consuming Cores, turning them into capturable, annotated errors.
+//
+// See https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/getsentry/sentry-go%24+file:%5Etransport%5C.go+defaultBufferSize&patternType=literal for more details.
+type worker struct {
+	// hub is an isolated Sentry context used to send out events to their API.
+	hub sentryHub
+	// C is the channel used to pass errors and their associated context to the go routine sending out events.
+	C chan *errorContext
+	// timeout tells the worker to wait wfpewfp
+	timeout chan struct{}
+	// done stops the worker from accepting new cores when written into.
+	done chan struct{}
+}
+
+type sentryHub struct {
+	hub *sentry.Hub
+	sync.Mutex
+}
+
+func (w *worker) setHub(hub *sentry.Hub) {
+	w.hub.Lock()
+	defer w.hub.Unlock()
+	w.hub.hub = hub
```

Comment:
super-nit, with a one-liner like this IMO it's easier to read if you just have the Unlock happen explicitly 😋 

```suggestion
	w.hub.hub = hub
	w.hub.Unlock()
```

Reply:
hahhaha, that's funny, for me it's the other way around, I'm so used to see a defer just after a lock that if that's not the case, my brain thinks there's a mistake :lau